core/src/main/java/org/jdbi/v3/core/ExtensionRegistry.java

```diff
@@ -55,9 +55,10 @@ static ExtensionRegistry copyOf(ExtensionRegistry registry) {
         this.factories = new CopyOnWriteArrayList<>();
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"}) // working around https://bugs.eclipse.org/bugs/show_bug.cgi?id=499197
     ExtensionRegistry(List<Entry<? extends ExtensionConfig<?>>> factories) {
-        this.factories = factories.stream()
-                .map(Entry::copyOf)
+        this.factories = (List) factories.stream()
+                .map(e -> Entry.copyOf((Entry)e))
```

Comment:
This is why I didn't bother generifying the `Entry` class originally. Sometimes it's just a ton of work making the compiler happy with generics.